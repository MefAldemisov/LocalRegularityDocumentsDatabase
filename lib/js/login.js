"use strict";

// baisic bootstrap validation TODO: write myself
".needs-validation".on("submit", function (event) {
  // event.preventDefault();
  if (form.checkValidity() === false) {
    event.preventDefault();
    event.stopPropagation();
  }

  form.classList.add('was-validated');
});