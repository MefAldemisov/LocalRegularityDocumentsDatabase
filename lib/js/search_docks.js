"use strict";

$("form").submit(function (event) {
  event.preventDefault();
});

function switch_filter_and_sort_forms() {
  $("#search_form").toggleClass("hidden");
  $("#filter_form").toggleClass("hidden");
}

$("#search_form_btn").click(function () {
  switch_filter_and_sort_forms();
});
$("#next_request_btn").click(function () {
  switch_filter_and_sort_forms();
});