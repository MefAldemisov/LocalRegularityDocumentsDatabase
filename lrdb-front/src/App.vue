<script>
// import "./node_modules/reset-css/reset.css";
import "./assets/css/main.css";
import "bootstrap-css-only";
import UI_logo from "./assets/images/IU_logo_black.png";
import BaseDropdown from "./components/search/input_items/BaseDropdown.vue"

export default {
    name: "App",
    props: {
        user_id: {
            type: Number,
            default: 23,
        },
        logout_link: {
            type: String,
            default: "#",
        },
        img_link: {
            type: String,
            default: UI_logo,
        },
    },
    data: function() {
        return {
            selected_lang: "EN",
            alt: "$t('alt_logo')",
            navs: ["my", "search", "load"],
            languages: [
                {val:"EN", text: "EN", index: 0}, 
                {val:"RU", text: "RU", index: 1},
                {val:"TAT", text: "TAT", index: 2}
            ]
        };
    },
    watch: {
        selected_lang:  function(new_l, old_l){
            this.$i18n.locale = new_l.toLowerCase();
            // change language of the web page
            document
                .getElementsByTagName("html")[0]
                .setAttribute("lang", this.$i18n.locale);
        }
    },
    methods: {
        changeTab: function(event){
            const current = this.$router.currentRoute.name;
            const index = this.navs.indexOf(current) === -1 ? 2 : this.navs.indexOf(current);
            
            if (event.srcKey ==="left"){
                console.log("true")
                this.$router.push({name : this.navs[(index-1) >= 0 ? (index - 1) : 2]});
            } else {
                this.$router.push({name : this.navs[(index+1)%this.navs.length]});
            }
        },
        goToMy: function(){
            this.$router.push({name : "my"});
        }
    },
    created: function() {
        // set language as default language of the user's browser
        const userLang = navigator.language || navigator.userLanguage;
        if (userLang[0] === "r") {
            this.selected_lang = "RU";
        } else {
            this.selected_lang = "EN";
        }
        this.$i18n.locale = this.selected_lang.toLowerCase();
    },
    components: {
        BaseDropdown
    }
};
</script>
<template>
    <div>
        <header>
            <nav class="symbolic">
                <ul
                    class="container d-flex flex-wrap justify-content-between align-items-baseline"
                >
                    <li class="burger clickbel">
                        <span>
                            <font-awesome-icon icon="bars"></font-awesome-icon>
                        </span>
                        <ul class="burger_list shadow-lg">
                            <router-link
                                class="dropdown-item rounded-lg"
                                active-class="burger_active"
                                tag="li"
                                v-for="nav in navs"
                                :to="{ name: nav }"
                                :key="nav"
                                >{{ $t(nav) }}</router-link
                            >
                            <li class="dropdown-item rounded-lg">
                                <a :href="logout_link" class="logout">
                                    {{ $t("logout") }}
                                </a>
                            </li>
                        </ul>
                    </li>
                    <li>
                        <img @click="goToMy" class="iu_logo" :src="img_link" :alt="alt" />
                    </li>
                    <router-link
                        tag="li"
                        v-for="nav in navs"
                        active-class="active"
                        class="clickbel navigation"
                        :to="{ name: nav }"
                        :key="nav"
                        >
                        {{ $t(nav) }}
                    </router-link>
                    <li class="clickbel navigation">
                        <a :href="logout_link" class="logout">
                            {{ $t("logout") }}
                        </a>
                         
                    </li>
                    <li class="selector">
                        <span class="active transparent">
                            H
                        </span>
                        <span class="navigation"
                            >{{ $t("lang_text") }} : 
                        </span>
                        <base-dropdown class="navigation lang-control"
                            v-model="selected_lang" 
                            :options="languages"
                        >
                        </base-dropdown>
                        </select>
                    </li>
                </ul>
            </nav>
        </header>
        <main class="container" v-shortkey="{right: ['ctrl','arrowright'], left: ['ctrl','arrowleft']}"  @shortkey="changeTab">
            <router-view></router-view>
        </main>
        <footer class="symbolic"></footer>
    </div>
</template>
<style>
.btn-success {
    /*UI style*/
    /* background-color: #15b012;  */
    /* Accessability low treshhold */
    color: #fff;
    background-color: #0f870d;
}
.bg-success {
    color: #fff;
    background-color: #0f870d !important; /*bootstrap uses important, thus to rewrite it...*/
}
.border-success {
    border-color: #0f870d;
}
.hidden {
    display: none;
}
</style>
<style scoped>
.symbolic {
    background-color: #edf1f5;
    min-height: 3.5rem;
}
footer {
    margin-top: 2rem;
}
.iu_logo {
    width: 8rem;
    cursor: pointer;
}
ul.container {
    width: 70%;
}
ul.container > li {
    height: 100%;
    padding: 0.5rem;
    display: inline-block;
}
.logout:link,
.logout:visited,
.logout:hover,
.logout:active {
    text-decoration: none;
}
.logout {
    color: inherit;
    pointer-events: none;
}
.navigation {
    font-size: 1rem;
    color: black;
}
.transparent {
    opacity: 0;
}
.active,
ul>li.clickbel:hover {
    color: black;
    font-weight: 900;
    font-size: 1.2rem;
    cursor: pointer;
    text-decoration: none;
}
main {
    padding-top: 1rem;
    margin: auto;
    width: 70%;
    min-height: 90vh;
    position: relative;
}
ul > li.burger {
    position: relative;
    display: none;
    padding: 0;
}
.burger_list {
    display: none;
    position: absolute;
    background-color: #fff;
    z-index: 4;
    padding: 12px 16px;
}
.burger_list > * {
    font-weight: 400;
}
.dropdown-item:active,
.burger_active {
    background-color: #0f870d;
    color: white;
}
select.custom-select {
    width: 50%;
    padding-left: 1rem;
}
.lang-control {
    display: inline-block;
    max-width: 4.3rem;
    min-width: 4.3rem;
    background-color: #edf1f5;
    border: none;
}
@media (max-width: 1200px) {
    main.container,
    ul.container {
        width: 90%;
        padding-right: 0;
        padding-left: 0;
    }
}
@media (max-width: 1000px) {
    ul > li.clickbel {
        display: none;
    }
    ul.container > li.burger {
        display: inline-block;
    }
    .burger:hover .burger_list,
    .burger:focus .burger_list {
        display: block;
    }
}
@media(min-width: 2000px) {
    main {
        min-height: 97vh;
    }
}
</style>
