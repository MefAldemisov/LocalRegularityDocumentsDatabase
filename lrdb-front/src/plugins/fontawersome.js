import Vue from "vue";
import { library } from "@fortawesome/fontawesome";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import {
    faSearch,
    faFile,
    faDownload,
    faHistory,
    faBars,
    faArrowRight,
    faArrowLeft,
} from "@fortawesome/fontawesome-free-solid";

library.add(
    faSearch,
    faFile,
    faDownload,
    faHistory,
    faBars,
    faArrowRight,
    faArrowLeft
);

Vue.component("font-awesome-icon", FontAwesomeIcon); // registered globally
