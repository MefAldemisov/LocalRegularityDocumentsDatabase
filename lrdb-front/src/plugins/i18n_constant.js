let en = require(`../assets/langs/en.json`);
let ru = require(`../assets/langs/ru.json`);
let tat = require(`../assets/langs/tat.json`);

export const i18n_obj = {
    locale: "en",
    fallbackLocale: "en", // time options
    messages: { en: en, ru: ru, tat: tat },
};
