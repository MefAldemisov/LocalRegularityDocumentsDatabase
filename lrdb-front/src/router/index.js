import Vue from "vue";
import VueRouter from "vue-router";

Vue.use(VueRouter);

function loadView(view) {
    return () => import(`../views/${view}.vue`);
}

const routes = [
    {
        path: "/my_docs",
        name: "my",
        component: loadView("MyDocsPage"),
    },
    {
        path: "/search",
        name: "search",
        component: loadView("SearchPage"),
    },
    {
        path: "/",
        name: "login",
        component: loadView("LoginPage"),
    },
    {
        path: "/*",
        redirect: "/",
        name: "login",
        component: loadView("LoginPage"),
    },
    {
        path: "/load",
        name: "load",
        component: loadView("LoadDocsPage"),
        redirect: "load/new",
        children: [
            {
                path: "/load/new",
                name: "new",
                component: loadView("load/LoadDocsPageNew"),
            },
            {
                path: "/load/change",
                name: "change",
                component: loadView("load/LoadDocsPageChange"),
            },
        ],
    },
    {
        path: "/history/:id",
        name: "History",
        component: loadView("HistoryPage"),
        props: true,
    },
];

const router = new VueRouter({
    mode: "history",
    base: process.env.BASE_URL,
    routes,
});

export default router;
