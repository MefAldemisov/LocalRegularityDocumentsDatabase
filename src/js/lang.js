// removes some browser issues
$.ajaxPrefilter(function (options, original_Options, jqXHR) {
    options.async = true
})
// variable, that tracs current language
let current_lang
// request for laguage dictionary + lang update
function getLanguage(lang_label) {
    $.ajax({
        url: 'js/langs/' + lang_label + '.json',
        dataType: 'json', async: false, dataType: 'json',
        success: function (language_dict) {
            Object.keys(language_dict).forEach(function (key) {
                $("#" + key).html(language_dict[key])
            })
            current_lang = lang_label
            $("html").attr("lang", current_lang)
        }
    })
}
// change laguage by click
$("#lang_marker").change(function () {
    getLanguage(this.value.toLowerCase())
})
