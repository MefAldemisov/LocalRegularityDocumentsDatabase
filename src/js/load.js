$(function () {
    // load the login page
    $("#content").load("login.html")
    // default language of the user's browser
    const userLang = navigator.language || navigator.userLanguage; 
    switch(userLang[0]) {
        case "r":
            current_lang = "ru"
            break
        default:
            current_lang = "en"
    }
    getLanguage(current_lang)
});
$(".clickbel").click(function () {
    $("#content").load(this.id + "_docks.html")
    getLanguage(current_lang)
})
