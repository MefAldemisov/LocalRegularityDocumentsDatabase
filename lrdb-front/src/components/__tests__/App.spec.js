import { shallowMount, createLocalVue } from "@vue/test-utils";
import App from "../../App.vue";
import VueI18n from "vue-i18n";
import VueRouter from "vue-router";
import { i18n_obj } from "../../plugins/i18n_constant.js";
import router from "../../router/index.js";
import "../../plugins/fontawersome.js";

const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(VueRouter);
localVue.use(require("vue-shortkey"), { prevent: ["input", "textarea"] });
let i18n = new VueI18n(i18n_obj);
describe(`App`, () => {
    // language is taken from navigator
    test.each([
        ["ru", "ru"],
        ["en", "en"],
        ["fr", "en"],
    ])(`initial language %s is taken from navigator`, async (lang, res) => {
        Object.defineProperty(window.navigator, "language", {
            value: lang,
            configurable: true,
        });
        Object.defineProperty(window.navigator, "userLanguage", {
            value: lang,
            configurable: true,
        });
        // const dom = new JSDOM();
        // global.document = dom.window.document;

        const w = shallowMount(App, { i18n, localVue, router });
        expect(w.vm.$i18n.locale).toBe(res);
        // await w.vm.$nextTick();
        // expect(
        //     global.document.getElementsByTagName("html")[0].getAttribute("lang")
        // ).toBe(res);
    });
    // language of html is updated with the paguage of user -- not testabe now

    // tab change
    describe(`tab change`, () => {
        const w = shallowMount(App, { i18n, localVue, router });
        w.vm.$router.push({ name: "my" });

        test.each([
            ["my", "right", "search"],
            ["search", "left", "my"],
            ["my", "left", "new"],
            ["new", "left", "search"],
            ["search", "right", "new"],
            ["new", "right", "my"],
        ])(`from '%s' to %s result '%s'`, async (current, dir, res) => {
            // w.vm.$router.push({ name: current });
            // await w.vm.$nextTick();
            expect(w.vm.$router.currentRoute.name).toBe(current);
            const event = { srcKey: dir };
            w.vm.changeTab(event);
            await w.vm.$nextTick();
            expect(w.vm.$router.currentRoute.name).toBe(res);
        });
    });

    describe(`go to My page`, () => {
        const w = shallowMount(App, { i18n, localVue, router });

        test.each([["my"], ["search"], ["new"]])(
            `from '%s'`,
            async (current) => {
                w.vm.$router.push({ name: current });
                await w.vm.$nextTick();
                expect(w.vm.$router.currentRoute.name).toBe(current);
                w.vm.goToMy();
                // await w.vm.$nextTick();
                expect(w.vm.$router.currentRoute.name).toBe("my");
            }
        );
    });

    // go to My
});
