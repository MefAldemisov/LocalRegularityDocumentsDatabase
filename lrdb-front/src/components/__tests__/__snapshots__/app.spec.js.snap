// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render to a snapshot when tips list has items 1`] = `
<aside>
  <load-tip-stub />
   
  <!---->
   
  <!---->
   
  <!---->
</aside>
`;

exports[`TodoList should render to a snapshot when tips list is empty 1`] = `
<aside>
  <load-tip-stub />
   
  <!---->
   
  <!---->
   
  <!---->
</aside>
`;
